### PR TITLE
Google OAuth Backend

### LINKED ISSUE #24: Login using Google Auth
**Describe the feature**
The user can log in to the WebApp using Google API.
Planing to use PassportJs and Google Strategy 2. 

**Screenshots if any**

**Device Information [optional]:**

**Additional context**

**Are you working on this issue? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by shrirangpdeshmukh**
The user can login using google.
The path is /auth/google/.
Issue number #24

### CODE REVIEW

**Comment by arjungarg07** [comment]
Have you tested this? @shrirangpdeshmukh

File: routes/index.js [hunk 1]
```diff
@@ -1,9 +1,34 @@
...
+// auth with google+
+router.get(
+  "/auth/google",
+  passport.authenticate("google", {
+    scope: ["profile", "email"],
+  })
+);
+
+// callback route for google to redirect to
+router.get(
```
Comment (arjungarg07) [inline routes/index.js:23]: Define all these routes in the controllers by making a new file of googleAuth.js, same as we have done for handling url controllers.

**Review by arjungarg07 — CHANGES_REQUESTED** [review]

**Comment by shrirangpdeshmukh** [comment]
> Have you tested this? @shrirangpdeshmukh

Yes I have tested this

File: routes/index.js (hunk 1 above)
Comment (shrirangpdeshmukh) [reply routes/index.js:23]: Okay

Comment (shrirangpdeshmukh) [reply routes/index.js:23]: Changes made as requested

File: controllers/googleAuth.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+function redirect(req, res) {
+  res.redirect("/");
+}
+
```
Comment (arjungarg07) [inline controllers/googleAuth.js:4]: We will be implementing a User dashboard so I think we should redirect it to that page only. What do you think?

Comment (shrirangpdeshmukh) [reply controllers/googleAuth.js:4]: Yes, I think it will be good.

File: config/default.json [hunk 3]
```diff
@@ -1,4 +1,7 @@
...
-	"mongoURI": "mongodb://localhost:27017/test",
-	"baseURL": "http://localhost:8000/"
-}
\ No newline at end of file
+  "mongoURI": "mongodb://localhost:27017/test",
+  "baseURL": "http://localhost:8000/",
+  "GOOGLE_CLIENT_ID": "144025185913-ireisec3jjgn1k05e4vdtgf0lrb2qpms.apps.googleusercontent.com",
+  "GOOGLE_CLIENT_SECRET": "hRbkQOz5sW_Wm52TGMwcs9wu",
+  "cookieKey": "secret-key-for-brief-me"
+}
```
Comment (arjungarg07) [inline config/default.json:4]: We should not expose these vital keys in the public repo.

**Review by arjungarg07 — CHANGES_REQUESTED** [review]

File: config/default.json (hunk 3 above)
Comment (shrirangpdeshmukh) [reply config/default.json:4]: I was thinking to make a .env file but was not sure how to share it to others.
Should I make one and share it on slack?

Comment (arjungarg07) [reply config/default.json:4]: Yeah, make an .env file and keep this info with you only, I will generate the credentials for deployment. Don't share it on slack. Thanks!

Comment (shrirangpdeshmukh) [reply config/default.json:4]: Done

**Comment by arjungarg07** [comment]
How are we maintaining a session here? Are we using check-session for a persistent session?

**Comment by shrirangpdeshmukh** [comment]
> How are we maintaining a session here? Are we using check-session for a persistent session?

We are using cookies for a persistent session

**Review by arjungarg07 — APPROVED** [review]

**Comment by arjungarg07** [comment]
@shrirangpdeshmukh, Right now if we are starting the server, It is throwing an error, because it is unable to find the configuration keys. What we want is that developers can start the server without any errors. Please, make the required changes so that no error gets thrown. Can we have a quick fix for this? Thanks!

**Comment by shrirangpdeshmukh** [comment]
@arjungarg07,
I can comment out the current auth implementation.
The authentication won't be working in that case, the google strategy requires the client details
Either I have to share env file with the developers or currently turn off the google authnetication.
What should I do?

**Pull request merged**